cunumeric/module.py

```diff
@@ -1392,6 +1399,233 @@ def squeeze(a: ndarray, axis: Optional[NdShapeLike] = None) -> ndarray:
+
+    Returns
+    -------
+    broadcast : array
+        A readonly view on the original array with the given shape.
+        It is typically not contiguous.
+        Furthermore, more than one element of a broadcasted array
+        may refer to a single memory location.
+
+    See Also
+    --------
+    numpy.broadcast_to
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    return _broadcast_to(arr, shape, subok)
+
+
+def _broadcast_arrays(
+    arrs: list[ndarray],
+    subok: bool = False,
+) -> list[ndarray]:
+    # create an arry object w/ options passed from 'broadcast' routines
+    arrays = list(array(arr, copy=False, subok=subok) for arr in arrs)
+    # check if the broadcast can happen in the input list of arrays
+    shapes = list(arr.shape for arr in arrays)
+    out_shape = broadcast_shapes(*shapes)
+    # broadcast to the final shape
+    arrays = list(_broadcast_to(arr, out_shape, subok) for arr in arrays)
+    return arrays
+
+
+def broadcast_arrays(
+    *args: Sequence[Any], subok: bool = False
+) -> list[ndarray]:
+    """
+
+    Broadcast any number of arrays against each other.
+
+    Parameters
+    ----------
+    `*args` : array_likes
+        The arrays to broadcast.
+
+    subok : bool, optional
+        If True, then sub-classes will be passed-through,
+        otherwise the returned arrays will be forced to
+        be a base-class array (default).
+
+    Returns
+    -------
+    broadcasted : list of arrays
+        These arrays are views on the original arrays.
+        They are typically not contiguous.
+        Furthermore, more than one element of a broadcasted array
+        may refer to a single memory location.
+        If you need to write to the arrays, make copies first.
+        While you can set the writable flag True,
+        writing to a single output value may end up changing
+        more than one location in the output array.
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+
+    """
+    arrs = list(convert_to_cunumeric_ndarray(arr) for arr in args)
+    return _broadcast_arrays(arrs, subok=subok)
+
+
+class broadcast:
+    def __init__(self, *arrays: Sequence[Any]) -> None:
+        """
+
+        Produce an object that mimics broadcasting.
+
+        Parameters
+        ----------
+        `*arrays` : array_likes
+            The arrays to broadcast.
+
+        subok : bool, optional
+            If True, then sub-classes will be passed-through,
+            otherwise the returned arrays will be forced to
+            be a base-class array (default).
+
+        Returns
+        -------
+        b : broadcast object
+            Broadcast the input parameters against one another,
+            and return an object that encapsulates the result.
+            Amongst others, it has shape and nd properties,
+            and may be used as an iterator.
+
+        """
+        arrs = list(convert_to_cunumeric_ndarray(arr) for arr in arrays)
```

Comment:
```suggestion
        arrs = [convert_to_cunumeric_ndarray(arr) for arr in arrays]
```

Reply:
@sbak5 : you still have `list(convert_to_cunumeric..` in module.py file for the methods you have added

Reply:
Done.